routers/docs.py

```diff
@@ -1,24 +1,100 @@
 import http.client
 import logging
 
-from fastapi import APIRouter, Depends, HTTPException
+from fastapi import APIRouter, BackgroundTasks, Depends, HTTPException
+
+from schemas.api import GenerateFileDocsRequest, GenerateFileDocsResponse, GetFileDocsResponse, DocsStatusEnum
+from schemas.database import FirestoreDocumentationCreateModel, FirestoreDocumentationUpdateModel
 
-from schemas.generated_doc import GenerateFileDocRequest, GeneratedDocResponse
 from services.documentation_service import DocumentationService, get_documentation_service
+from services.clients.firebase_client import FirebaseClient, get_firebase_client
 
 router = APIRouter()
 
+# background tasks for generating the documentation
+async def generate_documentation(
+    documentation_service: DocumentationService,
+    firebase_client: FirebaseClient,
+    firebase_file_id: str,
+    github_url: str, 
+):
+    generated_docs = await documentation_service.generate_doc_for_github_file(github_url)
+
+    blob_url = firebase_client.get_blob_url(firebase_file_id)
+
+    firebase_client.add_blob(blob_url, generated_docs.content)
+    
+    firebase_client.update_document(
+        FirebaseClient.TEST_COLLECTION, 
+        firebase_file_id, 
+        FirestoreDocumentationUpdateModel
+        (
+            bucket_url=blob_url,
+            status=DocsStatusEnum.COMPLETED
+        ).model_dump()
+    )
 
-@router.post("/file-doc")
-async def generate_file_doc(
-        generate_file_doc_request: GenerateFileDocRequest,
-        documentation_service: DocumentationService = Depends(get_documentation_service)
-) -> GeneratedDocResponse:
+
+@router.post("/file-docs")
+async def generate_file_docs(
+        generate_file_docs_request: GenerateFileDocsRequest,
+        background_tasks: BackgroundTasks,
+        firebase_client: FirebaseClient = Depends(get_firebase_client),
+        documentation_service: DocumentationService = Depends(get_documentation_service),
+) -> GenerateFileDocsResponse:
     try:
-        data = await documentation_service.generate_doc_for_github_file(file_url=generate_file_doc_request.url)
-        return data
+
+        # add document to firebase
+        document_ref = firebase_client.add_document(
```

Comment:
That makes sense. Does it make sense to leave the background_tasks call in lines 37-42 in here or move that out as well?

Reply:
Leave it in here. That is essentially our `3. Calling business logic`

Reply:
Ahh I think I see why you did this. Maybe this section of the Background Tasks doc could be useful:
https://fastapi.tiangolo.com/tutorial/background-tasks/#dependency-injection

So we call some method in DocumentationService and pass it the BackgroundTasks object. It might sort of look like this:
```python
def create_doc_job(background_tasks):
  # call add_document like we are doing here

  # add to background_tasks here

  # return the created document
```